addons/medical_engine/functions/fnc_getHitpointArmour.sqf

```diff
@@ -0,0 +1,73 @@
+#include "script_component.hpp"
+/*
+ * Author: Pterolatypus
+ * Checks a unit's equipment to calculate the total armour on a hitpoint
+ *
+ * Arguments:
+ * 0: Unit <OBJECT>
+ * 1: Hitpoint <STRING>
+ * 2: Disable Caching <BOOLEAN> (Default: false)
+ *
+ * Return Value:
+ * Total armour for the given hitpoint <ARRAY>
+ * 0: Armour <NUMBER>
+ * 1: ExplosionShielding <NUMBER>
+ *
+ * Example:
+ * [player, "HitChest"] call ace_medical_engine_fnc_getHitpointArmour
+ *
+ * Public: No
+ */
+
+params ["_unit","_hitpoint",["_noCache",false]];
+
+// Structural damage is inferred differently, not needed here
+if (_hitpoint in ["", "#structural"]) exitwith {[1,1]};
+
+
+// Get base values from the unit class
+private _unitCfg = configFile >> "CfgVehicles" >> (typeOf _unit);
+private _armour = if (isNumber (_unitCfg >> "armor")) then {getNumber (_unitCfg >> "armor")} else {2};
+private _explosion = if (isNumber (_unitCfg >> "explosionShielding")) then {getNumber (_unitCfg >> "explosionShielding")} else {0.4};
```

Comment:
Where do those hard coded numbers come from? Basic_Body? I feel like, since this is cached, it should read those from config instead of hard coding.

Reply:
These numbers are originally from "O_Soldier_F" because that's what I was looking at at the time
`armor=2` is the same for every unit in the game.
`explosionShielding` is either 0.4 (most opfor/indep) or 0.3 (most blufor)
So they seemed like sensible default values

Reply:
Well, I mean, you could fall back to the naked model:
```sqf
private _type = typeOf _unit; //"B_Soldier_F"
private _nakedClass = _cfgVehicles >> getText (_cfgWeapons >> getText (_cfgVehicles >> "B_Soldier_F" >> "nakedUniform"));
private _armor = getNumber (_nakedClass >> "armor");
private _explosion = getNumber (_nakedClass >> "explosionShielding");
```

However, assuming we can't guarantee that these values are defined, we can't be any more sure those have the values either. And the game likely wouldn't use them as fallback anyway. It would just go with 0 and possibly complain or act in a undefined manner (like missile projectile models without mass/zero mass).

Alternatively you could read them from CAManBase.

armor entry is first defined in class All. explosionShielding is first defined in CAManBase. CAManBase also defines armor with a sensible value. All objects inherit from CAManBase and if some mod for some reason created their own complete class, huge parts of the scripting wouldn't work as everyone is using CAManBase as common base class for person objects.

It should therefore really just be `getNumber`. We know the entry is defined or inherited.

Variable name should be spelled "_armor".

Reply:
Fair enough. I suppose, similar to the point below, if it isn't defined then something else is going to break first so it's not really our responsibility to check.

> Variable name should be spelled "_armor"

Dammit, thought I'd got away with that :P

Reply:
Changed every occurrence to `armor` for consistency